src/slider.jsx

```diff
@@ -323,17 +314,84 @@ const Slider = React.createClass({
       document.addEventListener('touchcancel', this._dragTouchEndHandler, false);
     }
     this._onDragStart(e);
+
+    // Cancel scroll and context menu
+    e.preventDefault();
   },
 
   _onHandleMouseDown(e) {
     if (document) {
       document.addEventListener('mousemove', this._dragHandler, false);
       document.addEventListener('mouseup', this._dragEndHandler, false);
-      this._toggleSelection('none');
+
+      // Cancel text selection
+      e.preventDefault();
+
+      // Set focus manually since we called preventDefault()
+      ReactDOM.findDOMNode(this.refs.handle).focus();
```

Comment:
It is true! I actually didn't look the React API for that, because in another part of the slider code we have: `ReactDOM.findDOMNode(this.refs.track).getBoundingClientRect().left;` that I believe is also unnecessary. So I agree we should change that!